dashboard/app/controllers/api/v1/pd/workshop_attendance_controller.rb

```diff
@@ -41,28 +34,10 @@ def show
   # PUT /api/v1/pd/workshops/1/attendance/:session_id/user/:user_id
   def create
     user_id = params[:user_id]
-    enrollment = nil
-
-    if new_attendance_model_enabled?
-      teacher = User.find(user_id)
-
-      # Attempt to find a matching enrollment
-      enrollment = @workshop.enrollments.find_by('user_id = ? OR email = ?', user_id, teacher.email)
-    else
-      # Legacy attendance.
-      # TODO(Andrew): remove once we settle on the new workflow.
-      teacher = @workshop.section.students.where(id: user_id).first
+    teacher = User.find(user_id)
 
-      # Admins can override and add a missing teacher to the section
-      if teacher.nil? && current_user.admin? && params[:admin_override]
-        # Still the teacher account must exist.
-        teacher = User.find_by!(id: user_id, user_type: User::TYPE_TEACHER)
-        @workshop.section.add_student teacher
-      end
-
-      # renders a 404 (not found)
-      raise ActiveRecord::RecordNotFound.new('teacher required') unless teacher
-    end
+    # Attempt to find a matching enrollment
+    enrollment = @workshop.enrollments.find_by('user_id = ? OR email = ?', user_id, teacher.email)
```

Comment:
Thinking through edge base - will this ever be nil? If not, suggest making this find_by so if it is, we blow up at this line and not line 43

Reply:
Good point, presuming you meant `find_by!`. It shouldn't be nil. It also won't blow up on line 43, it will just set the enrollment to nil. But I like the idea of blowing up here for a bad user id. I'll add that.